documentation/src/main/asciidoc/topics/proc_cdi_injecting_embedded.adoc

```diff
@@ -0,0 +1,93 @@
+[id='cdi_inject_embed']
+= Injecting Embedded Caches
+Set up CDI beans to inject embedded caches.
+
+.Procedure
+
+. Create a cache qualifier annotation.
++
+[source,java]
+----
+...
+import javax.inject.Qualifier;
+
+@Qualifier
+@Target({ElementType.FIELD, ElementType.PARAMETER, ElementType.METHOD})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface GreetingCache { <1>
+}
+----
++
+<1> creates a `@GreetingCache` qualifier.
++
+. Add a producer class that defines cache configuration objects.
++
+[source,java]
+----
+...
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.cdi.ConfigureCache;
+import javax.enterprise.inject.Produces;
+
+public class Config {
+
+    @ConfigureCache("mygreetingcache") <1>
+    @GreetingCache <2>
+    @Produces
+    public Configuration greetingCacheConfiguration() {
+        return new ConfigurationBuilder()
+                    .memory()
+                        .size(1000)
+                    .build();
+    }
+}
+----
++
+<1> names the cache to inject.
+<2> adds the cache qualifier.
++
+. Configure transport for clustered cache modes, if required.
++
+[source,java]
+----
+...
+package org.infinispan.configuration.global.GlobalConfigurationBuilder;
+
+public class Config {
+
+    @GreetingCache
```

Comment:
I'd add a note here too, saying smth like 

```
<1> adds the cache qualifier. Normally an application does not need more than one cache manager, and the cache manager producer should have no qualifier.
```

Or maybe it's better to mention this in the existing note 2?